MENTAL HEALTH SURVEY 2025

About Dataset
This dataset compiles self-reported information from individuals to examine how professional background, work habits, stress patterns, and personal coping mechanisms influence mental health and emotional well-being. The data highlights the complex relationship between workplace engagement, lifestyle behavior, and mental resilience.

In [2]:
import numpy as np 
import pandas as pd 

In [4]:
df = pd.read_csv(r"C:\Users\ayush\Desktop\datasets\Mental Health dataset1.csv")
df

,Gender,Country,Occupation,SelfEmployed,FamilyHistory,Treatment,DaysIndoors,HabitsChange,MentalHealthHistory,IncreasingStress,MoodSwings,SocialWeakness,CopingStruggles,WorkInterest,SocialWeakness.1,MentalHealthInterview,CareOptions
0,Female,UK,Others,No,Yes,No,15-30 days,No,Yes,Yes,High,No,Yes,Maybe,No,No,No
1,Female,USA,Housewife,No,Yes,No,15-30 days,Maybe,Maybe,Yes,High,Maybe,Yes,Maybe,Maybe,No,Not sure
2,Female,Canada,Others,No,No,Yes,More than 2 months,Maybe,No,No,Medium,No,No,No,No,No,Not sure
3,Female,Canada,Corporate,No,Yes,Yes,Go out Every day,No,Maybe,No,Medium,No,No,Maybe,No,No,Not sure
4,Female,UK,Corporate,No,No,Yes,More than 2 months,Yes,No,Maybe,Medium,Maybe,No,Maybe,Maybe,No,Not sure
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
261323,Male,USA,Business,No,No,No,15-30 days,Maybe,No,No,Low,Maybe,Yes,No,Maybe,Maybe,Not sure
261324,Male,USA,Business,Yes,Yes,Yes,15-30 days,Maybe,No,No,Low,Maybe,Yes,No,Maybe,Maybe,Not sure
261325,Male,USA,Business,No,Yes,No,15-30 days,Maybe,No,No,Low,Maybe,Yes,No,Maybe,No,No
261326,Male,USA,Business,No,Yes,Yes,15-30 days,Maybe,No,No,Low,Maybe,Yes,No,Maybe,No,Yes


In [6]:
print(df.columns)

Index(['Gender', 'Country', 'Occupation', 'SelfEmployed', 'FamilyHistory',
       'Treatment', 'DaysIndoors', 'HabitsChange', 'MentalHealthHistory',
       'IncreasingStress', 'MoodSwings', 'SocialWeakness', 'CopingStruggles',
       'WorkInterest', 'SocialWeakness.1', 'MentalHealthInterview',
       'CareOptions'],
      dtype='str')


In [9]:
df.shape


(261328, 17)

In [11]:
df.head()

,Gender,Country,Occupation,SelfEmployed,FamilyHistory,Treatment,DaysIndoors,HabitsChange,MentalHealthHistory,IncreasingStress,MoodSwings,SocialWeakness,CopingStruggles,WorkInterest,SocialWeakness.1,MentalHealthInterview,CareOptions
0,Female,UK,Others,No,Yes,No,15-30 days,No,Yes,Yes,High,No,Yes,Maybe,No,No,No
1,Female,USA,Housewife,No,Yes,No,15-30 days,Maybe,Maybe,Yes,High,Maybe,Yes,Maybe,Maybe,No,Not sure
2,Female,Canada,Others,No,No,Yes,More than 2 months,Maybe,No,No,Medium,No,No,No,No,No,Not sure
3,Female,Canada,Corporate,No,Yes,Yes,Go out Every day,No,Maybe,No,Medium,No,No,Maybe,No,No,Not sure
4,Female,UK,Corporate,No,No,Yes,More than 2 months,Yes,No,Maybe,Medium,Maybe,No,Maybe,Maybe,No,Not sure


In [12]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 261328 entries, 0 to 261327
Data columns (total 17 columns):
 #   Column                 Non-Null Count   Dtype
---  ------                 --------------   -----
 0   Gender                 261328 non-null  str  
 1   Country                261328 non-null  str  
 2   Occupation             261328 non-null  str  
 3   SelfEmployed           256972 non-null  str  
 4   FamilyHistory          261328 non-null  str  
 5   Treatment              261328 non-null  str  
 6   DaysIndoors            261328 non-null  str  
 7   HabitsChange           261328 non-null  str  
 8   MentalHealthHistory    261328 non-null  str  
 9   IncreasingStress       261328 non-null  str  
 10  MoodSwings             261328 non-null  str  
 11  SocialWeakness         261328 non-null  str  
 12  CopingStruggles        261328 non-null  str  
 13  WorkInterest           261328 non-null  str  
 14  SocialWeakness.1       261328 non-null  str  
 15  MentalHealthInterview  26132

# DATA CLEANING 

All Values are non empty except self employed 

In [13]:
df.isnull().sum()  # give  where exact no of missing value

Gender                      0
Country                     0
Occupation                  0
SelfEmployed             4356
FamilyHistory               0
Treatment                   0
DaysIndoors                 0
HabitsChange                0
MentalHealthHistory         0
IncreasingStress            0
MoodSwings                  0
SocialWeakness              0
CopingStruggles             0
WorkInterest                0
SocialWeakness.1            0
MentalHealthInterview       0
CareOptions                 0
dtype: int64

As missing ratio is small we simply drop them for data integrity 

So SelfEmployed has missing values around 4k+

In [ ]:
df = df.dropna(subset=['SelfEmployed'])

In [ ]:
df.isna().sum()

Gender                   0
Country                  0
Occupation               0
SelfEmployed             0
FamilyHistory            0
Treatment                0
DaysIndoors              0
HabitsChange             0
MentalHealthHistory      0
IncreasingStress         0
MoodSwings               0
SocialWeakness           0
CopingStruggles          0
WorkInterest             0
SocialWeakness.1         0
MentalHealthInterview    0
CareOptions              0
dtype: int64

In [20]:
df.shape

(256972, 17)

Removed all missing values for the dataset


In [19]:
(df['SocialWeakness'] == df['SocialWeakness.1']).sum()

np.int64(256972)

SocialWeakness.1 is a duplicate column so we will remove it just for removing redundancy and make data set clean and professional 

In [21]:
df = df.drop(columns=['SocialWeakness.1'])

In [23]:
df.head()

,Gender,Country,Occupation,SelfEmployed,FamilyHistory,Treatment,DaysIndoors,HabitsChange,MentalHealthHistory,IncreasingStress,MoodSwings,SocialWeakness,CopingStruggles,WorkInterest,MentalHealthInterview,CareOptions
0,Female,UK,Others,No,Yes,No,15-30 days,No,Yes,Yes,High,No,Yes,Maybe,No,No
1,Female,USA,Housewife,No,Yes,No,15-30 days,Maybe,Maybe,Yes,High,Maybe,Yes,Maybe,No,Not sure
2,Female,Canada,Others,No,No,Yes,More than 2 months,Maybe,No,No,Medium,No,No,No,No,Not sure
3,Female,Canada,Corporate,No,Yes,Yes,Go out Every day,No,Maybe,No,Medium,No,No,Maybe,No,Not sure
4,Female,UK,Corporate,No,No,Yes,More than 2 months,Yes,No,Maybe,Medium,Maybe,No,Maybe,No,Not sure


A duplicate column (SocialWeakness.1) was identified and removed after confirming it contained identical values to SocialWeakness.

In [24]:
for col in df.columns:
    print(col, df[col].unique())

Gender <StringArray>
['Female', 'Male']
Length: 2, dtype: str
Country <StringArray>
[                    'UK',                    'USA',                 'Canada',
            'New Zealand',              'Australia',                 'Poland',
           'South Africa',                'Ireland',                'Belgium',
            'Netherlands',                  'India',                 'Sweden',
                 'Brazil', 'Bosnia and Herzegovina',               'Colombia',
             'Costa Rica',                'Croatia',         'Czech Republic',
                'Denmark',                'Finland',                 'France',
                'Georgia',                'Germany',                 'Greece',
                 'Israel',                  'Italy',                 'Mexico',
                'Moldova',                'Nigeria',            'Philippines',
               'Portugal',                 'Russia',              'Singapore',
            'Switzerland',               'Thail

In [25]:
for cols in df.columns: 
    df[cols] = df[cols].astype(str).str.lower().str.strip()


In [26]:
for col in df.columns:
    print(col, df[col].unique())

Gender <StringArray>
['female', 'male']
Length: 2, dtype: str
Country <StringArray>
[                    'uk',                    'usa',                 'canada',
            'new zealand',              'australia',                 'poland',
           'south africa',                'ireland',                'belgium',
            'netherlands',                  'india',                 'sweden',
                 'brazil', 'bosnia and herzegovina',               'colombia',
             'costa rica',                'croatia',         'czech republic',
                'denmark',                'finland',                 'france',
                'georgia',                'germany',                 'greece',
                 'israel',                  'italy',                 'mexico',
                'moldova',                'nigeria',            'philippines',
               'portugal',                 'russia',              'singapore',
            'switzerland',               'thail

Data was standardized using lowercase formatting to ensure consistency. For presentation purposes, values were later formatted appropriately

# Now Getting Exploratory Data Analysis

In [27]:
df.head()

,Gender,Country,Occupation,SelfEmployed,FamilyHistory,Treatment,DaysIndoors,HabitsChange,MentalHealthHistory,IncreasingStress,MoodSwings,SocialWeakness,CopingStruggles,WorkInterest,MentalHealthInterview,CareOptions
0,female,uk,others,no,yes,no,15-30 days,no,yes,yes,high,no,yes,maybe,no,no
1,female,usa,housewife,no,yes,no,15-30 days,maybe,maybe,yes,high,maybe,yes,maybe,no,not sure
2,female,canada,others,no,no,yes,more than 2 months,maybe,no,no,medium,no,no,no,no,not sure
3,female,canada,corporate,no,yes,yes,go out every day,no,maybe,no,medium,no,no,maybe,no,not sure
4,female,uk,corporate,no,no,yes,more than 2 months,yes,no,maybe,medium,maybe,no,maybe,no,not sure


In [28]:
df['IncreasingStress'].value_counts()

IncreasingStress
maybe    89195
yes      86923
no       80854
Name: count, dtype: int64

The high number of ‘maybe’ responses suggests uncertainty and lack of awareness, meaning many individuals may not fully recognize or acknowledge their stress levels.

Does Work Interest Creates Stress ?

In [29]:
pd.crosstab(df['WorkInterest'],df['IncreasingStress'],normalize= 'index')

IncreasingStress,maybe,no,yes
WorkInterest,,,
maybe,0.299569,0.352836,0.347596
no,0.407048,0.294064,0.298888
yes,0.329110,0.295167,0.375723


Individuals with low work interest show the highest proportion of uncertainty in stress levels, indicating disengagement may contribute to mental instability 


Interestingly, individuals with high work interest also report significant stress levels, suggesting that stress is not solely dependent on interest but may be influenced by workload or external pressures

Is Staying Indoor creates stress ?


In [30]:
pd.crosstab(df['DaysIndoors'],df['IncreasingStress'],normalize='index')

IncreasingStress,maybe,no,yes
DaysIndoors,,,
1-14 days,0.342958,0.315868,0.341174
15-30 days,0.331072,0.317982,0.350946
31-60 days,0.299547,0.357319,0.343135
go out every day,0.382506,0.323274,0.294220
more than 2 months,0.382675,0.253204,0.364120


Individuals spending more than two months indoors show the highest levels of stress, indicating a strong link between prolonged isolation and declining mental well-being.

People who go out daily report the lowest stress levels, suggesting that regular outdoor activity may help reduce stress.



Moderate indoor duration (31–60 days) shows relatively balanced stress levels, suggesting that extreme isolation, rather than indoor time itself, is a key contributor to stress.

Is Coping cause stress?

In [31]:
pd.crosstab(df['CopingStruggles'], df['IncreasingStress'], normalize='index')

IncreasingStress,maybe,no,yes
CopingStruggles,,,
no,0.364751,0.300120,0.335129
yes,0.326956,0.331214,0.341830


Coping struggles do not show a strong direct correlation with increasing stress, as stress levels remain relatively similar across both groups.

is family history affect treatment ?

In [32]:
pd.crosstab(df['FamilyHistory'], df['Treatment'], normalize='index')

Treatment,no,yes
FamilyHistory,,
no,0.653391,0.346609
yes,0.274511,0.725489


Individuals with a family history of mental health issues are significantly more likely to seek treatment (72.5%) compared to those without such history (34.7%), indicating that prior exposure increases awareness and willingness to take action.

Do care options affect whether people take treatment?

In [34]:
pd.crosstab(df['CareOptions'], df['Treatment'], normalize='index')

Treatment,no,yes
CareOptions,,
no,0.603889,0.396111
not sure,0.617365,0.382635
yes,0.300564,0.699436


Access to or awareness of care options significantly increases the likelihood of seeking treatment, with nearly 70% of individuals opting for treatment when care options are available

In [35]:
df.to_csv("cleaned_mental_healthsurvey_data.csv", index=False)

# Recommendations

1. Increase Awareness of Mental Health Resources
Organizations should actively promote available mental health support systems, as lack of awareness is a major barrier to treatment.

2. Encourage Work-Life Balance and Outdoor Activities
Reducing prolonged indoor stays through flexible work policies and encouraging social or outdoor engagement can help lower stress levels.

3. Provide Accessible Mental Health Support Systems
Companies should implement counseling services, employee assistance programs, and easy access to mental health professionals.

4. Target High-Risk Groups for Early Intervention
Individuals showing uncertainty (“maybe” responses) should be prioritized for awareness and support programs, as they represent a vulnerable group.

# Conclusion

This analysis reveals that mental health challenges are widespread and influenced more significantly by lifestyle factors such as prolonged isolation than by work-related engagement alone. Among all factors, access to and awareness of mental health care options emerged as the most critical driver of treatment-seeking behavior. The findings highlight the importance of not only providing mental health support but also ensuring that individuals are well-informed and encouraged to utilize these resources. Addressing these areas can lead to improved well-being and more proactive mental health management